socorro/processor/rules/mozilla.py

```diff
@@ -288,6 +290,67 @@ def action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
         processed_crash["java_stack_trace"] = java_stack_trace
 
 
+class MalformedBreadcrumbs(Exception):
+    pass
+
+
+def validate_breadcrumbs(data):
+    """Validates a Breadcrumbs data structure
+
+    :arg list data: list of breadcrumbs dicts
+
+    :raises: MalformedBreadcrumbs if it's malformed in some way
+
+    """
+    # NOTE(willkg): Sentry doesn't _require_ timestamp in their Breadcrumbs event, but
+    # it's highly recommended. We try to match their spec since that seemed helpful.
+    # The other top level fields are optional. If we want to require other fields, we
+    # can add them here.
+    required_keys = {"timestamp"}
+
+    if not isinstance(data, list):
+        raise MalformedBreadcrumbs("not a list")
+
+    for i, item in enumerate(data):
+        if not isinstance(item, dict):
+            raise MalformedBreadcrumbs(f"item {i} not a dict")
+
+        # missing_keys is the required_keys minus the intersection of required_keys and
+        # the item's keys
+        missing_keys = required_keys - (required_keys & set(item.keys()))
+        if missing_keys:
+            missing_keys = ", ".join(sorted(missing_keys))
+            raise MalformedBreadcrumbs(f"item {i} missing keys: {missing_keys}")
```

Comment:
At some point, I want to do a bunch of schema stuff and when I do that, we'll redo this with that.